src/main/java/org/opentripplanner/standalone/config/NodeAdapter.java

```diff
@@ -335,13 +342,14 @@ public void logUnusedParameters(Logger log) {
     public boolean equals(Object o) {
         if (this == o) { return true; }
         if (o == null || getClass() != o.getClass()) { return false; }
-        NodeAdapter adapter = (NodeAdapter) o;
-        return source.equals(adapter.source);
+        NodeAdapter other = (NodeAdapter) o;
+        return Objects.equals(source, other.source)
+                && Objects.equals(contextPath, other.contextPath);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(source);
+        return Objects.hash(source, contextPath);
```

Comment:
This does correct the behavior of equals and hashcode, but since we're putting them in a list we probably don't even need semantic equality. Less stuff to maintain if we can determine they are not even needed...